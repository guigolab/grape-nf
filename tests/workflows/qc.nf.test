nextflow_workflow {

    name "Test Workflow QC"
    script "workflows/qc.nf"
    workflow "QC"

    tag "workflow"

    test("Should run BAM QC (compressed annotation)") {

        when {
            params {
                inferExpTool = "RSeQC"
                bamStatsTool = "bamstats"
                annotation = "${baseDir}/data/annotation.gtf.gz"
            }
            workflow {
                """
                input[0] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true]
                ])
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("Should run BAM QC (uncompressed annotation)") {

        when {
            params {
                inferExpTool = "RSeQC"
                bamStatsTool = "bamstats"
                annotation = "${baseDir}/data/annotation.gtf"
            }
            workflow {
                """
                input[0] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true]
                ])
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toTranscriptome.bam"), "bam", "TranscriptomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
