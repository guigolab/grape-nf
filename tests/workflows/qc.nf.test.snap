{
    "Should run BAM QC (uncompressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome.bam:md5,2fea3e65b19a08afbf9b1279e97fe7e0",
                        "bam",
                        "GenomeAlignments",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toTranscriptome.bam:md5,3a8316414e15048a188e532fd51216c9",
                        "bam",
                        "TranscriptomeAlignments",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "2": [
                    [
                        "sample3",
                        "test3",
                        "sample3_stats.json:md5,77c72bef6f1410e33c90dbc01697fffc",
                        "json",
                        "BamStats",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "bamStats": [
                    [
                        "sample3",
                        "test3",
                        "sample3_stats.json:md5,77c72bef6f1410e33c90dbc01697fffc",
                        "json",
                        "BamStats",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genomeAlignments": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome.bam:md5,2fea3e65b19a08afbf9b1279e97fe7e0",
                        "bam",
                        "GenomeAlignments",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "transcriptomeAlignments": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toTranscriptome.bam:md5,3a8316414e15048a188e532fd51216c9",
                        "bam",
                        "TranscriptomeAlignments",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-19T17:01:24.605245"
    },
    "Should run BAM QC (compressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome.bam:md5,2fea3e65b19a08afbf9b1279e97fe7e0",
                        "bam",
                        "GenomeAlignments",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toTranscriptome.bam:md5,3a8316414e15048a188e532fd51216c9",
                        "bam",
                        "TranscriptomeAlignments",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "2": [
                    [
                        "sample3",
                        "test3",
                        "sample3_stats.json:md5,77c72bef6f1410e33c90dbc01697fffc",
                        "json",
                        "BamStats",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "bamStats": [
                    [
                        "sample3",
                        "test3",
                        "sample3_stats.json:md5,77c72bef6f1410e33c90dbc01697fffc",
                        "json",
                        "BamStats",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "genomeAlignments": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome.bam:md5,2fea3e65b19a08afbf9b1279e97fe7e0",
                        "bam",
                        "GenomeAlignments",
                        true,
                        "MATE2_SENSE"
                    ]
                ],
                "transcriptomeAlignments": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toTranscriptome.bam:md5,3a8316414e15048a188e532fd51216c9",
                        "bam",
                        "TranscriptomeAlignments",
                        true,
                        "MATE2_SENSE"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-19T17:01:53.25532"
    }
}